pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -100,54 +86,230 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
-	// Define a new Pod object
-	pod := newPodForCR(instance)
-
-	// Set GroupPermission instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
+	// get list of clusterRole on k8s
+	clusterRoleList := &v1.ClusterRoleList{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, clusterRoleList)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		// error reading the object - requeue the request
 		return reconcile.Result{}, err
 	}
 
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
-		if err != nil {
-			return reconcile.Result{}, err
+	// get permission type, empty at first
+	permission := &managedv1alpha1.Permission{}
+
+	// get clusterRoleName from permission type
+	clusterRoleName := permission.ClusterRoleName
+
+	// if crClusterRoleNameList returns true with values in there
+	allClusterRoleNameInCluster, crClusterRoleNameList := checkClusterRoleName(instance, clusterRoleList)
+	if allClusterRoleNameInCluster == false {
+		for _, crClusterRoleName := range crClusterRoleNameList {
+			// get the conditions array
+			groupPermissionConditions := instance.Status.Conditions
+
+			// make new condition
+			newCondition := managedv1alpha1.Condition{
+				LastTransitionTime: metav1.Now(),
+				Message:            "ClusterRole for clusterPermission does not exist",
+				ClusterRoleName:    crClusterRoleName,
+				Status:             true,
+				State:              "Failed",
+			}
+
+			// append new condition to conditions array
+			groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+			// update GroupPermission object with new conditions
+			instance.Status.Conditions = groupPermissionConditions
+
+			err = r.client.Status().Update(context.TODO(), instance)
```

Comment:
TODO: decide if we want one batch update or single condition that includes names of all missing clusterroles.